### PR TITLE
fix: Fix the test of _mm_h[add|sub]s_epi16

### PR DESCRIPTION
**Pull request by marktwtn**
Add the checking of saturated minimum value for _mm_hadds_epi16
and saturated maximum value for _mm_hsubs_epi16.

### CODE REVIEW

**Comment by marktwtn** [comment]
@jserv could you test the branch of this pull request on Apple M1?

**Comment by jserv** [comment]
> @jserv could you test the branch of this pull request on Apple M1?

Ask @junwei0117 for experimenting.

File: tests/impl.cpp [hunk 1]
```diff
@@ -6449,39 +6449,35 @@ result_t test_mm_hadd_pi32(const SSE2NEONTestImpl &impl, uint32_t i)
 result_t test_mm_hadds_epi16(const SSE2NEONTestImpl &impl, uint32_t i)
 {
     int16_t s16_max = 0x7fff;
+    int16_t s16_min = 0x8000;
```
Comment (jserv) [inline tests/impl.cpp:6452]: Append some comments for `s16_{max,min}`, such as the motivations.

Comment (marktwtn) [reply tests/impl.cpp:6452]: I decide to remove the variables and replace them with macro `INT16_MAX` and `INT16_MIN`.

**Pull request merged**